src/applications/common/schemaform/save-in-progress/helpers.js

```diff
@@ -0,0 +1,59 @@
+import {
+  createPageList,
+  createFormPageList,
+  createRoutes
+} from '../helpers';
+import RoutedSavablePage from './RoutedSavablePage';
+import RoutedSavableReviewPage from './RoutedSavableReviewPage';
+import FormSaved from './FormSaved';
+import SaveInProgressErrorPage from './SaveInProgressErrorPage';
+
+export function createRoutesWithSaveInProgress(config) {
+  const protectedRoutes = new Set(['introduction', 'review-and-submit', 'confirmation', '*']);
+  const formPages = createFormPageList(config);
+  const pageList = createPageList(config, formPages);
+  const newRoutes = createRoutes(config);
+
+  newRoutes.forEach((route, index) => {
+    let newRoute;
+
+    // rewrite page component
+    if (!protectedRoutes.has(route.path)) {
+      newRoute = Object.assign({}, route, { component: RoutedSavablePage });
+      newRoutes[index] = newRoute;
+    }
+
+    // rewrite review page component
+    if (route.path === 'review-and-submit') {
+      newRoute = Object.assign({}, route, { component: RoutedSavableReviewPage });
+      newRoutes[index] = newRoute;
+    }
+  });
+
+  if (!config.disableSave) {
+    // const lengthOfRoutes = newRoutes.length;
+    // const positionToAddRoutes = lengthOfRoutes - 1;
+
+    newRoutes.splice(newRoutes.length - 1, 0, {
+      path: 'form-saved',
+      component: FormSaved,
+      pageList,
+      config
```

Comment:
Changing this from `formConfig` to `config` makes me a little nervous. The non-protected routes still expect and use `formConfig`. I think this might trip us up later.

Reply:
no worries, I'll change it back! also noticing I have some commented out lines of code above, will remove that too